### DISCUSSION TITLE
Is there plans for supporting custom subjects when using `com.atproto.moderation.createReport`

### DISCUSSION THREAD

**Discussion by citorva** [bluesky-social/atproto#3257] [Bluesky Lexicons]
Hello, using domain specific labelers such as XBlock, I think that it would be interesting to allow labelers to define custom subjects.

**Comment by bnewbold** [comment]
Could you clarify what you mean? The term "subject" in the context of moderation reports means the account (DID) or specific piece of content (AT-URI) being reported. With the current APIs, any DID and any record type can be reported.

**Reply by imax9000** [reply]
Most likely it's about customizing `reasonType`

**Reply by citorva** [reply]
You are right, I am interested in allowing services implementing the `createReport` method to provide custom available `reasonType` instead of the static enum.

I think that some services providing *moderation* features on specific use cases can be interested in this feature.

**Comment by citorva** [comment]
Hello, my message is about customizing ReasonType. Currently, the protocol only allow a set of reasons which are not necessary adapted to needs of some moderation tools. My first idea is to provide endpoints enumerating available reasons.

**Reply by imax9000** [reply]
IMO, a more straightforward approach would be to list them in the `app.bsky.labeler.service` record directly.
